mesheryctl/internal/cli/root/root.go

```diff
@@ -121,23 +123,74 @@ func init() {
 
 // initConfig reads in config file and ENV variables if set.
 func initConfig() {
+	// Allow user to override config file with use of --config global flag
 	if cfgFile != "" {
 		// Use config file from the flag.
 		viper.SetConfigFile(cfgFile)
+		// Otherwise, use the default `config.yaml` config file
 	} else {
-		// Use default ".meshery" folder location.
-		if _, err := os.Stat(utils.MesheryFolder); err != nil {
-			if os.IsNotExist(err) {
-				err = os.MkdirAll(utils.MesheryFolder, 0775)
-				if err != nil {
-					log.Fatal(err)
+		if _, err := os.Stat(fmt.Sprintf("%s/%s", utils.MesheryFolder, "config.yaml")); os.IsNotExist(err) {
+			log.Printf("Missing Meshery config file.")
+			userResponse := utils.AskForConfirmation("Create default config now?")
+
+			// Check for Meshery existence and permission of application folder
+			if userResponse {
+				if _, err := os.Stat(utils.MesheryFolder); err != nil {
+					if os.IsNotExist(err) {
+						err = os.MkdirAll(utils.MesheryFolder, 0775)
+						if err != nil {
+							log.Fatal(err)
+						}
+					}
+				}
+
+				// Create Meshery config file using standard template
```

Comment:
This function should be defined only once. There should be only one definition of the standard template.

Let's consolidate the two functions between root.go and create.go.

Reply:
I dont get what you mean by this ? Create two functions ( initConfig and add/editAdapter ) ?